src/components/windows/types/agenda/editAgendaEntry.jsx

```diff
@@ -0,0 +1,166 @@
+import { useForm } from "react-hook-form";
+import { InnerContainer, InnerContainerRow, InputButton, InputCheckbox, InputContainer, InputElement, InputLabel, LabelWarning, PanelButton } from "../../../dashboard"
+import { Agenda, getCurrentEvent } from '@phoenixlan/phoenix.js'
+import { FormButton, FormInput } from "../../../form";
+import { useContext, useEffect, useState } from "react";
+import { Notice } from "../../../containers/notice";
+import { AuthenticationContext } from "../../../authentication";
+
+export const EditAgendaEntry = ({functions, entries}) => {
+    const { register, handleSubmit, watch, formState: { errors } } = useForm();
+
+    const [ title, setTitle ]											= useState(entries.title);
+    const [ description, setDescription ]								= useState(entries.description);
+    const [ time, setTime]												= useState(entries.time ? new Date(entries.time ? entries.time*1000+7200000 : undefined).toISOString().slice(0, 16) : undefined)
+    const [ location, setLocation ]										= useState(entries.location);
+	const [ deviatingTime, setDeviatingTime ]							= useState(entries.deviating_time ? new Date(entries.deviating_time ? entries.deviating_time*1000+7200000 : undefined).toISOString().slice(0, 16) : undefined);
+	const [ deviatingLocation, setDeviatingLocation ]					= useState(entries.deviating_location);
+	const [ deviatingTimeUnknown, setDeviatingTimeUnknown ] 			= useState(entries.deviating_time_unknown);
+	const [ deviatingInformation, setDeviatingInformation]				= useState(entries.deviating_information);
+	const [ statePinned, setStatePinned ] 								= useState(entries.pinned);
+	const [ stateCancelled, setStateCancelled ] 						= useState(entries.cancelled);
+	const [ stateDeviatingControl, setStateDeviatingControl]			= useState(undefined);
+
+
+    useEffect(() => {
+		/*
+			onLoad_CheckDeviatingControl:
+			Input fields like deviating location, deviating time and deviating information is normally
+			disabled to prevent the user from using the wrong fields. We use this function to check if
+			the database has deviating information, if so, enable the fields on load.
+		*/
+		const onLoad_CheckDeviatingControl = () => {
+			if(deviatingTime || deviatingLocation || deviatingInformation || deviatingTimeUnknown) {
+				setStateDeviatingControl(true);
+			}
+		}
+
+		onLoad_CheckDeviatingControl();
+    }, []);
+
+    const onSubmit = async (data) => {
+        // Get current event so we can get its UUID
+        const event = await getCurrentEvent();
+
+		// If stateDeviatingControl is set to false, prevent the following variables to submit data.
+		if(!stateDeviatingControl) {
+			data.deviating_time = null;
```

Comment:
`undefined` unsets them, `null` will still send the value `null` across the network. Is this intentional?

Reply:
This is intentional.
We want to change the variables to `null` when we uncheck the second checkbox which says "Deviating entry" and submit. This is because when you uncheck this, we say that the entry is no longer deviating, all deviating information can be cleared.

If we use undefined, none of the variables will change, and will stay the same, which we dont want.